vault/resource_generic_secret_test.go

```diff
@@ -96,13 +143,36 @@ func testResourceGenericSecret_initialCheck(expectedPath string) resource.TestCh
 		}
 
 		client := testProvider.Meta().(*api.Client)
-		secret, err := client.Logical().Read(path)
-		if err != nil {
-			return fmt.Errorf("error reading back secret: %s", err)
+		var secret *api.Secret
+		var data map[string]interface{}
+		if isV2 {
+			secretList, err := client.Logical().List("secretsv2/metadata")
```

Comment:
`List()` returns a `Secret`, may be clearer  to rename to `secretList` to `resp` since you are shadowing `secret`